### ISSUE TITLE
More examples

### ISSUE THREAD

**Issue by opcodewriter** [vkhorikov/CSharpFunctionalExtensions#519]
Vlad, thanks for your great library!

Maybe it's just me but I'm having a hard time understanding how to use the library in the correct way, otherwise chance is, I will write code which has potential bugs.

Beside the README, the only resource I could find is your post "[Functional C#: Handling failures, input errors](https://enterprisecraftsmanship.com/posts/functional-c-handling-failures-input-errors/)" but it looks outdated, there's no "OnSuccess" method anymore.

**Comment by vkhorikov** [comment]
Yeah, docs are lacking. One way is to use the `OnSuccess` from the course and then follow the deprecation warnings, they say which method you need to use instead.

But I personally don't use use extension methods much (aside from the very basic `Map` and `Bind`) and still get a lot of value from the lib on my projects -- from the `Result`, `Maybe`, and `ValueObject` vanilla classes.
